### PR TITLE
add middleware for token validation

### PR DESCRIPTION
**Pull request by cvillamer13**

### CODE REVIEW

File: src/middlewares/token.validation.middleware.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    getAuth()
+        .verifyIdToken(token)
+        .then(( decryptToken : any) => {
+            next()
+        })
+        .catch((e) => {
+            res.send(400)
+        })
+    
+
```
Comment (aldenvallestero) [inline src/middlewares/token.validation.middleware.ts:17]: move your **next()** here. It should always be the last command for all middlewares.

File: src/middlewares/token.validation.middleware.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+export function tokenValidation(req : Request, res : Response, next : NextFunction){
+    const authHeader = req.headers['authorization'];
+    const token = authHeader.split(' ')[1];
+    getAuth()
+        .verifyIdToken(token)
+        .then(( decryptToken : any) => {
+            next()
+        })
+        .catch((e) => {
+            res.send(400)
```
Comment (aldenvallestero) [inline src/middlewares/token.validation.middleware.ts:14]: make it return res.send(400)

File: src/middlewares/token.validation.middleware.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import { getAuth } from 'firebase-admin/auth';
+
+
+export function tokenValidation(req : Request, res : Response, next : NextFunction){
+    const authHeader = req.headers['authorization'];
+    const token = authHeader.split(' ')[1];
+    getAuth()
+        .verifyIdToken(token)
+        .then(( decryptToken : any) => {
+            next()
```
Comment (aldenvallestero) [inline src/middlewares/token.validation.middleware.ts:11]: remove your next() here.

File: src/middlewares/token.validation.middleware.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { Request, Response, NextFunction } from 'express'
+import { getAuth } from 'firebase-admin/auth';
+
+
+export function tokenValidation(req : Request, res : Response, next : NextFunction){
+    const authHeader = req.headers['authorization'];
+    const token = authHeader.split(' ')[1];
+    getAuth()
```
Comment (aldenvallestero) [inline src/middlewares/token.validation.middleware.ts:8]: get the userId and pass it to req.event.userId so you can use it anywhere

File: src/middlewares/token.validation.middleware.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import { Request, Response, NextFunction } from 'express'
+import { getAuth } from 'firebase-admin/auth'
+
+export interface UIDRequest extends Request {
```
Comment (aldenvallestero) [inline src/middlewares/token.validation.middleware.ts:4]: Anong kababalaghan to.

File: src/middlewares/token.validation.middleware.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  userID: any
+}
+
+export async function tokenValidation(
+  req: UIDRequest,
```
Comment (aldenvallestero) [inline src/middlewares/token.validation.middleware.ts:9]: req: Request

File: src/middlewares/token.validation.middleware.ts (hunk 5 above)
Comment (cvillamer13) [reply src/middlewares/token.validation.middleware.ts:4]: hindi updated iyan

File: src/middlewares/token.validation.middleware.ts (hunk 6 above)
Comment (cvillamer13) [reply src/middlewares/token.validation.middleware.ts]: luma na iyan mag mga bagong push ako

File: src/controllers/profile.controller.ts [hunk 7]
```diff
@@ -10,7 +10,9 @@ import {
   updateProfileIntroduction as UpdateProfileIntroduction,
 } from '../services/profile.service'
 
-export async function createProfile(req: Request, res: Response) {
+export async function createProfile(req: Request, res: Response, next : NextFunction) {
```
Comment (aldenvallestero) [inline src/controllers/profile.controller.ts:13]: Bakit may next dito, controller to ah.

File: src/middlewares/token.validation.middleware.ts [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+  req: Request,
+  res: Response,
+  next: NextFunction
+) {
+  const authHeader: any = req.headers['authorization']
+  const token = authHeader.split(' ')[1]
+  getAuth()
+    .verifyIdToken(token)
+    .then((decryptToken: any) => {
+      res.locals.userID = decryptToken.uid
```
Comment (aldenvallestero) [inline src/middlewares/token.validation.middleware.ts:15]: res.userId =

**Pull request merged**